rotors_gazebo_plugins/src/gazebo_imu_plugin.cpp

```diff
@@ -211,8 +260,18 @@ void GazeboImuPlugin::addNoise(Eigen::Vector3d* linear_acceleration,
 
 }
 
-// This gets called by the world update start event.
+
 void GazeboImuPlugin::OnUpdate(const common::UpdateInfo& _info) {
+
+  if(kPrintOnUpdates) {
+    gzdbg << __FUNCTION__ << "() called." << std::endl;
+  }
+
+  if(!pubs_and_subs_created_) {
+    CreatePubsAndSubs();
```

Comment:
This was a problem I had a while back. I used to have publisher/subscriber creation in `Load()` but I ran into a problem in where the ROS interface plugin would then sometimes miss "connection" messages. I think that because the plugin order is somewhat undefined, if say, the IMU plugin's `Load()` gets called first, and publishes a "connect" message to the ROS interface plugin, the ROS interface plugin will miss this if it has not yet be initialised. 

I don't think messages queue if there is no subscriber, I think they just get dropped. My solution was to put the pub/sub creation into the `Run()` method, as the ROS interface plugin is guaranteed to be running at this point. Can you think of a better solution?